core_v2/sources/domains.move

```diff
@@ -679,7 +685,7 @@ module aptos_names_v2::domains {
         let record = get_record_mut(domain_name, option::some(subdomain_name));
         assert!(option::is_some(&record.subdomain_ext), error::invalid_state(ENOT_A_SUBDOMAIN));
             let subdomain_ext = option::borrow(&record.subdomain_ext);
-            if (subdomain_ext.use_domain_expiration_sec) {
+            if (subdomain_ext.subdomain_expiration_policy == SUBDOMAIN_POLICY_LOOKUP_DOMAIN_EXPIRATION) {
                 assert!(false, error::invalid_state(ESUBDOMAIN_IS_AUTO_RENEW));
```

Comment:
if auto-renew flag is on and user wants to set the  expiration date, we can either
1. let them set and implicitly switch the flag back
2. throw error because the flag is on

I like the second one better because there's no extra action implied. I don't feel strong on either one though